servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/GrpcInstrumentationHelper.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    /**
+     * Creates a gRPC client instrumentation helper.
+     *
+     * @param openTelemetry the OpenTelemetry instance
+     * @param options OpenTelemetry configuration options
+     * @param componentName component name for peer service attribute
+     * @return client instrumentation helper
+     */
+    static GrpcInstrumentationHelper createClient(OpenTelemetry openTelemetry, OpenTelemetryOptions options,
+                                                  String componentName) {
+        SpanNameExtractor<RequestInfo> clientSpanNameExtractor = GrpcSpanNameExtractor.INSTANCE;
+        InstrumenterBuilder<RequestInfo, GrpcTelemetryStatus> clientInstrumenterBuilder =
+                Instrumenter.builder(openTelemetry, INSTRUMENTATION_SCOPE_NAME, clientSpanNameExtractor);
+        clientInstrumenterBuilder
+                .setSpanStatusExtractor(GrpcSpanStatusExtractor.INSTANCE)
+                .addAttributesExtractor(new DeferredGrpcClientAttributesExtractor(options));
+
+        if (options.enableMetrics()) {
+            clientInstrumenterBuilder.addOperationMetrics(HttpClientMetrics.get());
+        }
+        componentName = componentName.trim();
+        if (!componentName.isEmpty()) {
+            clientInstrumenterBuilder.addAttributesExtractor(
+                    AttributesExtractor.constant(PEER_SERVICE, componentName));
+        }
+
+        Instrumenter<RequestInfo, GrpcTelemetryStatus> instrumenter =
+                clientInstrumenterBuilder.buildClientInstrumenter(RequestHeadersPropagatorSetter.INSTANCE);
+
+        return new GrpcInstrumentationHelper(instrumenter, true);
+    }
+
+    /**
+     * Determines if a request should be treated as a gRPC request.
+     * <p>
+     * A request is considered gRPC if the Content-Type header starts with "application/grpc"
+     *
+     * @param request the HTTP request to examine
+     * @return true if this should be treated as a gRPC request, false for HTTP
+     */
+    boolean isGrpcRequest(StreamingHttpRequest request) {
+        CharSequence contentType = request.headers().get(CONTENT_TYPE);
+        return contentType != null && startsWithPrefix(contentType);
+    }
+
+    // TODO: maybe there is a more efficient way to do this comparison.
+    private static boolean startsWithPrefix(CharSequence charSequence) {
+        int seqLength = charSequence.length();
+        if (seqLength < GRPC_CONTENT_TYPE.length()) {
+            return false;
+        }
+        if (seqLength == GRPC_CONTENT_TYPE.length()) {
+            return GRPC_CONTENT_TYPE.contentEquals(charSequence);
+        }
+        // Start at the end since many content types start with 'application/' so we're more likely
+        // to be able to abort early by checking that the prefix ends with 'grpc'.
+        for (int i = GRPC_CONTENT_TYPE.length() - 1; i >= 0; i--) {
+            if (charSequence.charAt(i) != GRPC_CONTENT_TYPE.charAt(i)) {
```

Comment:
I'm afraid the content-type values are case-insensitive. For that we have handy utilities in `CharSequences`, like `equalsIgnoreCaseLower` for chars and `contentEqualsIgnoreCase` for `CharSequence`. If you go that route, it will be worth switching to `newAsciiString` for `GRPC_CONTENT_TYPE` constant